specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2023-12-01-preview/SystemComponents.json

```diff
@@ -0,0 +1,217 @@
+          {
+            "$ref": "../../common/2022-03-01/parameters.json#/parameters/ClusterResourceNameParameter"
+          },
+          {
+            "$ref": "../../common/2022-03-01/parameters.json#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "systemComponents",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SystemComponent"
+            },
+            "description": "Properties necessary to Create an System component."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request received successfully",
+            "schema": {
+              "$ref": "#/definitions/SystemComponent"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "SystemComponents",
+          "ClusterSystemComponents"
+        ],
+        "description": "Gets Kubernetes Cluster SystemComponents.",
+        "operationId": "SystemComponents_Get",
+        "x-ms-examples": {
+          "Get System component": {
+            "$ref": "./examples/GetSystemComponent.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../common/2022-03-01/parameters.json#/parameters/ClusterRpParameter"
+          },
+          {
+            "$ref": "../../common/2022-03-01/parameters.json#/parameters/ClusterResourceNameParameter"
+          },
+          {
+            "$ref": "../../common/2022-03-01/parameters.json#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SystemComponent"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Component": {
```

Comment:
Could this name be more specific in case we have non-system components? Maybe something like `SystemComponentInstance`?